bot/exts/events/advent_of_code/views/dayandstarview.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+
+import discord
+
+AOC_DAY_ANDSTAR_TEMPLATE = "{rank: >4} | {name:25.25} | {completion_time: >10}"
+
+
+class AoCDropdownView(discord.ui.View):
+    """Represents a Dropdown to filter AoC stats by Day and Star."""
+
+    def __init__(self, original_author: discord.Member, day_and_star_data: dict[str: dict], maximum_scorers: int):
+        super().__init__()
+        self.day = 1
+        self.star = 1
+        self.data = day_and_star_data
+        self.element_map = {element.custom_id: element for element in self.children}
```

Comment:
Why is this used now?

Reply:
You mean the `element_map`? If so, I came up with this, because I couldn't find an alternative to the built in `children` attribute, I mean that as in using them with indexes. If there's a better alternative I would be more than interested to implement it.

Reply:
But you have `self.star_select` no?